src/keywrap/keyprovider.rs

```diff
@@ -0,0 +1,572 @@
+// Copyright The ocicrypt Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{anyhow, Result};
+use crate::keywrap::KeyWrapper;
+use crate::utils::keyprovider as keyproviderpb;
+use tonic;
+use std::collections::hash_map::RandomState;
+use crate::config::{DecryptConfig, EncryptConfig, KeyProviderAttrs, Command};
+use std::collections::HashMap;
+use crate::{utils};
+use tokio;
+use crate::utils::{CommandExecuter};
+use tokio::runtime::Runtime;
+use core::option::Option;
+use core::fmt::Debug;
+use tonic::codegen::http::Uri;
+
+
+impl Debug for dyn CommandExecuter {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(f, "CommandExecuter")
+    }
+}
+
+/// A KeyProvider keywrapper
+#[derive(Debug)]
+pub struct KeyProviderKeyWrapper {
+    pub provider: String,
+    pub attrs: KeyProviderAttrs,
+    pub runner: Option<Box<dyn CommandExecuter>>,
+}
+
+pub const OP_KEY_WRAP: &str = "keywrap";
+pub const OP_KEY_UNWRAP: &str = "keyunwrap";
```

Comment:
Having an enum, against which you implement the `Display` trait, would be more robust:

```Rust
enum OpKey {
   Wrap,
   Unwrap,
}

impl fmt::Display for OpKey {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match *self {
            OpKey::Wrap => write!(f, "keywrap"),
            OpKey::Unwrap => write!(f, "keyunwrap"),
        }
    }
```

With `Display`, you can call `to_string()` on an `OpKey` and get an owner string back.

Reply:
done